### ISSUE TITLE
how: new (sub)section (How To's)

### ISSUE THREAD

**Issue by jorgeorpinel** [iterative/dvc.org#899]
It seems from several open issues that we may need a section or subsection somewhere in the docs to provide certain "how to" guides. For example, these are the issues with "how to" in their titles atm:

> UPDATE: See https://documentation.divio.com/how-to-guides/#how-to

- [x] Organize existing how-to material into the new section (some may be under guides now).
- [x] explain **how to** "undo" 'dvc add' + other "**how to** undo mistakes" recipes #625
- [x] commit: add example on **how to** add missing dependency to existing pipeline #612
- [ ] how to update tracked directories (part of #3297)
- [ ] https://github.com/iterative/dvc.org/issues/1079 ?

Other ideas:

- search old "how to" questions in `dvc` repo e.g. https://github.com/iterative/dvc/issues/599, forum, etc.
- cmd ref: improve `run` and `commit` / **how to** add output without re-running stage? #460
- "**How to** remove data" — what other ways to remove data (other than un-doing `add`) are there?
- "**How to** host a public DVC project" (private-write, public-read)
- ~~"**How to** uninstall dvc" for both binary and pip packages~~ #65
- This https://github.com/iterative/dvc.org/pull/732#pullrequestreview-306762834 discussion:
  > we'll have to introduce a how to section soon... May be we can move things like "Running on Windows" into this section
- Also, this https://github.com/iterative/dvc.org/pull/868#pullrequestreview-338379858 comment:
  > **Parallel stage execution**  from the `dvc repro` cmd ref can be probably moved to a **How to** section, with a link to it from the cmd ref
- How tos on `dvc init --no-scm` usage? See https://github.com/iterative/dvc.org/issues/100#issuecomment-657783092

Let's analyze this list, look for any others, and see whether it makes sense to have a dedicated how-to section, or whether an existing section of the docs e.g. use-cases or specific command references or user guides can cover all of these.

**Comment by jorgeorpinel** [comment]
💡 If we do need this new section, perhaps it could live near the new **Troubleshooting** section (in **User Guide**).

**Comment by PseudoNerd** [comment]
Can we expand the troubleshooting section by researching and discovering known errors and beginner mistakes while using DVC to make a list, and then analyze the formed list to make a dedicated how-to page with information pulled from the docs for better access? 


I'd like to participate in the first project for this year's edition of GSoD and DVC is something that I have used and had some experience with.

**Comment by jorgeorpinel** [comment]
Hey @PseudoNerd!

This epic issue is more about a proposed How to section (to contain several pages, each one a how-to) and less about the troubleshooting guide which is a single page where we plan to add more known problems. Both things need some work indeed and/or maybe they could be merged. Perhaps a good topic for a GSoD proposal.

For now I would encourage you to investigate smaller docs issues to get more familiar with DVC and our process 🙂 

I'll email you a longer reply about GSoD.

**Comment by MisterAwesome23** [comment]
Hey, 
My name is Manthan Admane and I am in my senior year of Computer Engineering undergrad.
I would like to work on "How To section" project idea under GSoD2020 and contribute to the community.

As suggested:
> For now I would encourage you to investigate smaller docs issues to get more familiar with DVC and our process

I'll start with getting familiar with fixing and exploring smaller issues on DVC issues. 
Looking forward to it.

**Comment by jorgeorpinel** [comment]
Hi @MisterAwesome23 OK. Please email your GSoD intro again to info@dvc.org so we have it on record. I'll send you an email with guidance over there. Best

**Comment by PseudoNerd** [comment]
While going through the docs and going through the issues that could make up the contents of this section, I realized that the stack overflow page for dvc can add to this list of potential how-tos. I made [this](https://stackoverflow.com/questions/tagged/dvc) search to find a handful of questions that could be helpful in making a how-to guide. More than 20 different questions were found out of which some are mentioned [here](https://github.com/iterative/dvc.org/issues/899#issue-545536339). 


> Let's analyze this list, look for any others, and see whether it makes sense to have a dedicated how-to section, or whether an existing section of the docs e.g. use-cases or specific command references or user guides can cover all of these.

If there is to be a dedicated how-to section in the docs then it should reflect the order of the levels present in the existing sections and have them internally linked to the specific how-to. Furthermore, a single how-to question(guide) will often refer to a set of dvc commands and use cases, so having them linked internally is a more comprehensive way of going through the documentation which has already been laid out. 
This is why it seems fitting to have a dedicated section for the how-to guide after the command reference section of the docs.
This section could have a link to it in the troubleshooting page as well. 

The pages in the tutorials and use-cases section can be absorbed into multiple `how-tos` by extracting relevant information. 

For a much simpler example, 

Q. How to track files and directories in dvc? 
Ans. You can use the dvc add command to start tracking files while using dvc. [Here](http://dvc.org/doc/command-reference/add#example-single-file) is an example of how to track a file and [here](http://localhost:8000/doc/command-reference/add#example-directory) is how you can track a directory in dvc by using the add command.


I went through various youtube links on seminars by Dmitry Petrov on youtube to understand the core of DVC so that I could understand dvc better. I'm relatively new to dvc and I feel that to be able to build a more comprehensive how-to section, communication with the developers and the community will be very helpful.

**Comment by jorgeorpinel** [comment]
> I made this search (https://stackoverflow.com/questions/tagged/dvc) to find a handful of questions that could be helpful in making a how-to guide.

Good idea. We can also review recent Monthly Gems blog posts e.g. https://dvc.org/blog/may-20-community-gems. But these small/specific questions are not enough for how-tos. We would need to find patterns among them and extrapolate larger how-to guides that cover most of these.

> a dedicated how-to section in the docs then it should reflect the order of the levels...

I don't know if I understood this explanation. Maybe something more graphical proposal or even a draft PR to showcase your suggestion would be better.

> For a much simpler example,
> Q. How to track files and directories in dvc?
> Ans...

This looks more like a F.A.Q. Again, I think we need more complex scenarios for how-to versions, each one with a page. Many proposals are already listed in the description of this issue.

However @PseudoNerd, this epic is still too big right now, as mentioned in https://github.com/iterative/dvc.org/issues/899#issuecomment-628324853. It would be great to have a bit more experiences working with you on smaller issues and circle back on this a bit later during the GSoD proposal stage.

> communication with the developers and the community will be very helpful

You can always ask questions and get help on Discord. You should have all the info. you need in your email but please ping me otherwise.

**Comment by jorgeorpinel** [comment]
> a dedicated how-to section in the docs then it should reflect the order of the levels...

I don't know if I understood this explanation. Maybe something more graphical proposal or even a draft PR to showcase your suggestion would be better.

> For a much simpler example,
> Q. How to track files and directories in dvc?
> Ans...

This looks more like a F.A.Q. Again, I think we need more complex scenarios for how-to versions, each one with a page. Many proposals are already listed in the description of this issue.

However @PseudoNerd, this epic is still too big right now, as mentioned in https://github.com/iterative/dvc.org/issues/899#issuecomment-628324853. It would be great to have a bit more experiences working with you on smaller issues and circle back on this a bit later during the GSoD proposal stage.

> communication with the developers and the community will be very helpful

You can always ask questions and get help on Discord. You should have all the info. you need in your email but please ping me otherwise.

**Comment by SidharthBansal** [comment]
Hi, I am interested in doing this for Google Season of Docs 2020. I have contributed to Public Lab in some freelancing contracts, Google Code In administrative services, Google Summer of Code 2018 Internship and Google Summer of Code 2019 mentorship in Ruby on Rails, HTML, CSS and Javascript. I have also done internship at Google, Hyderabad in android. I have knowledge of C++ & Java too. I have published one research paper in the field on NLP and other research paper is under review. I was department rank 1 in my branch in my college DTU, India.

I will start with pushing some minor documentation patches soon. Thanks
